datalad_metalad/extract.py

```diff
@@ -310,11 +310,13 @@ def __call__(
         # requested and the extractor class is a subclass of
         # DatasetMetadataExtractor (or a legacy extractor class).
         if path:
+            extractor_type = 'file'
             # Check whether the path points to a sub_dataset.
             ensure_path_validity(source_dataset, file_tree_path)
-            yield from do_file_extraction(extraction_arguments)
         else:
-            yield from do_dataset_extraction(extraction_arguments)
+            extractor_type = 'dataset'
+        
+        yield from do_extraction(ep=extraction_arguments, extractor_type=extractor_type)
```

Comment:
```suggestion
            extractor_type = 'dataset'
        
        yield from do_extraction(ep=extraction_arguments)
```
(add extractor_type to ExtractionArguments)